jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/awsrules/ec2/AssetTypeGroupedVulnerabilitiesRule.java

```diff
@@ -116,16 +116,18 @@ public PolicyResult execute(Map<String, String> ruleParam, Map<String, String> r
     private String getVMVulnerabilityDetails(List<JsonObject> vulnerabilityInfoList, String severity) {
         List<VulnerabilityInfo> vulnerabilityList = new ArrayList<>();
         for (JsonObject vulnerability : vulnerabilityInfoList) {
-            for (JsonElement cveDetails : vulnerability.get(severity).getAsJsonArray()) {
-                VulnerabilityInfo vul = new VulnerabilityInfo();
-                String cveId = cveDetails.getAsJsonObject().get("id").getAsString();
-                CveDetails cve = new CveDetails(cveId, PacmanUtils.NIST_VULN_DETAILS_URL + cveId);
-                List<CveDetails> cveList = new ArrayList<>();
-                cveList.add(cve);
-                vul.setCveList(cveList);
-                vul.setVulnerabilityUrl(cveDetails.getAsJsonObject().get("url").getAsString());
-                vul.setTitle(cveDetails.getAsJsonObject().get("title").getAsString());
-                vulnerabilityList.add(vul);
+            if (vulnerability.get(severity) != null) {
```

Comment:
In what situation the severity is null? We need to treat the issue at the source.

What would UI show when this is skipped?

Reply:
@plyubich  For each severity policy, the respective CVEs are fetched, and if there are no CVEs with medium severity for example, this will throw NPE, vulnerability.get(severity).getAsJsonArray() . I believe this is the source which needs an NPE check because the doc from vulnerability index may not contain CVEs for a certain severity.